tests/acquire_shard_test.go

```diff
@@ -102,7 +103,23 @@ func TestAcquireShard_OwnershipLostErrorSuite(t *testing.T) {
 // SetupSuite reads the shard ownership lost error fault injection config from the testdata folder.
 func (s *OwnershipLostErrorSuite) SetupSuite() {
 	s.AcquireShardSuiteBase.SetupSuite()
-	s.FunctionalTestBase.SetupSuiteWithCluster("testdata/acquire_shard_ownership_lost_error.yaml")
+	s.FunctionalTestBase.SetupSuiteWithCluster(testcore.WithFaultInjectionConfig(
+		config.FaultInjection{
+			Targets: config.FaultInjectionTargets{
+				DataStores: map[config.DataStoreName]config.FaultInjectionDataStoreConfig{
+					config.ShardStoreName: {
+						Methods: map[string]config.FaultInjectionMethodConfig{
+							"UpdateShard": {
+								Errors: map[string]float64{
+									"ShardOwnershipLost": 1.0,
```

Comment:
just thinking out loud...

should we have some kind of builder helper for these to simplify the code? e.g.
```
var fi config.FaultInjectionConfig
fi.Add(config.ShardStoreName, "UpdateShard", "ShardOwnershipLost", 1.0)
// fi.SetSeed(config.ShardStoreName, "UpdateShard", 43)
s.FunctionalTestBase.SetupSuiteWithCluster(testcore.WithFaultInjectionConfig(fi)
```